label_studio/users/models.py

```diff
@@ -95,7 +95,13 @@ class User(UserMixin, AbstractBaseUser, PermissionsMixin, UserLastActivityMixin)
     is_active = models.BooleanField(
         _('active'),
         default=True,
-        help_text=_('Designates whether to treat this user as active. ' 'Unselect this instead of deleting accounts.'),
+        help_text=_('Designates whether to treat this user as active. Unselect this instead of deleting accounts.'),
+    )
+    is_deleted = models.BooleanField(
+        _('deleted'),
+        default=False,
+        db_index=True,
```

Comment:
I'm not sure indexing this field like this will give us any performance gains

Maybe it's worth investigating the usecase more, but I think we would be okay without an index.

Reply:
Yeah we might be ok without an index but given the user table will filter on this field by default on every user query (is_deleted=False), I think it can't hurt and can only help even if the gain isn't huge - even small gains will add up on less db load. 

To your point, most user queries query on an individual user (generally by id or email) so I'm guessing in those cases, the index wont make a difference.  Also we expect most users will not be in deleted state so this feidl will be weighted False which is primarily what we're querying on which also would suggest it would not effect performance as much.

I think I was leaning towards using an index simply from the mindset that it doesn't hurt us much to add it (boolean indexes are relatively lightweight) and since its a field we'll always be querying on it feels safer add it.  Also to protect from unexpected performance issues from new user queries (for example if we add a query for is_deleted=True).

If you feel strongly about this I can remove it but for now I'll leave it in.